From d0b87ee87b282e9b91a1af924050e217b0b2ae8b Mon Sep 17 00:00:00 2001
From: Eric Haszlakiewicz <erh+git@nimenees.com>
Date: Mon, 12 Aug 2019 00:30:45 +0000
Subject: [PATCH] Add an explicit cast to double to squash a
 -Wimplicit-int-float-conversion warning. Though we will no longer be
 comparing exactly against INT64_MAX, this is ok because any value of that
 magnitude stored in a double will *also* have been rounded up, so the
 comparison will work appropriately.

Index: json_object.c
--- json_object.c.orig
+++ json_object.c
@@ -698,7 +698,9 @@ int64_t json_object_get_int64(const struct json_object
 	case json_type_int:
 		return jso->o.c_int64;
 	case json_type_double:
-		if (jso->o.c_double >= INT64_MAX)
+		// INT64_MAX can't be exactly represented as a double
+		// so cast to tell the compiler it's ok to round up.
+		if (jso->o.c_double >= (double)INT64_MAX)
 			return INT64_MAX;
 		if (jso->o.c_double <= INT64_MIN)
 			return INT64_MIN;
